### ISSUE TITLE
Failing to change cursor and display --REPLACE-- mode when I hit "r" to enter replace mode.

### ISSUE THREAD

**Issue by guhan-balaji** [VSCodeVim/Vim#6211]
**Describe the bug**

**To Reproduce**

1.  Go to '...'
2.  Click on '....'
3.  Scroll down to '....'
4.  See error

**Expected behavior**

**Screenshots**
If remapping-related, please attach log output: https://github.com/VSCodeVim/Vim#debugging-remappings.

**Environment (please complete the following information):**

- Extension (VsCodeVim) version:
- VSCode version:
- OS:

**Additional context**

**Comment by facundobatista** [comment]
Yes, please, the cursor when doing `r` should change like when doing `R`, as vim does. Thanks!!

**Comment by facundobatista** [comment]
I've been exploring the code a little and differentiate the cursor for `r` and `R` is not that easy.

We're handling "r + anychar" as a single action, so we don't control how the cursor looks "after `r` but before `anychar`:
```
export class ActionReplaceCharacter extends BaseCommand {
  modes = [Mode.Normal];
  keys = ['r', '<character>'];
  ...
```

We could have something like a transient mode, and one hand do:

```
@RegisterAction
export class CommandReplaceAtCursorFromNormalMode extends BaseCommand {
  modes = [Mode.Normal];
  keys = ['r'];

  public override async exec(position: Position, vimState: VimState): Promise<void> {
    await vimState.setCurrentMode(Mode.OneCharReplace);
  }
}
```

... and then ...

```
@RegisterAction
export class ActionReplaceCharacter extends BaseCommand {
  modes = [Mode.OneCharReplace];
  keys = ['<character>'];
  override createsUndoPoint = true;
  override runsOnceForEachCountPrefix = false;

  public override async exec(position: Position, vimState: VimState): Promise<void> {
    const toReplace = this.keysPressed[0];
    ...
```

**BUT**

- I wouldn't know how to get the `vimState.recordedState.count` before the `r` in `ActionReplaceCharacter`
- This all gets complicated as `r` can be hit also in visual mode and in visualblock mode
- Surely there are a zillion other things to consider that I don't know as I never looked into this code before

Thanks for all anyway! Hope this can be achievable.
